src/lib/plugin.js

```diff
@@ -32,70 +32,104 @@ module.exports = class PluginVirtual extends EventEmitter2 {
     assertOptionType(opts, 'currencyCode', 'string')
     assertOptionType(opts, 'currencyScale', 'number')
     assertOptionType(opts, 'maxBalance', 'string')
-    assertOptionType(opts, 'secret', 'string')
-    assertOptionType(opts, 'peerPublicKey', 'string')
-    assertOptionType(opts, '_store', 'object')
-    assertOptionType(opts, 'rpcUri', 'string')
-
-    this._secret = opts.secret
-    this._peerPublicKey = opts.peerPublicKey
-    this._publicKey = Token.publicKey(this._secret)
+
+    if (opts.rpcUris) {
+      assertOptionType(opts, 'rpcUris', 'object')
+      this._rpcUris = opts.rpcUris
+    } else {
+      assertOptionType(opts, 'rpcUri', 'string')
+      this._rpcUris = [ opts.rpcUri ]
+    }
+
     this._currencyCode = opts.currencyCode.toUpperCase()
     this._currencyScale = opts.currencyScale
 
-    this._store = opts._store
-    this._maxBalance = opts.maxBalance
-    this._balance = new Balance({
-      maximum: this._maxBalance,
-      store: this._store
-    })
+    if (opts.token && opts.prefix) {
+      assertOptionType(opts, 'token', 'string')
+      assertOptionType(opts, 'prefix', 'string')
 
-    // give a 'balance' event on balance change
-    this._balance.on('balance', (balance) => {
-      this.emit('balance', balance)
-    })
-
-    // Token uses ECDH to get the ledger prefix from secret and public key
-    this._authToken = Token.authToken({
-      secretKey: this._secret,
-      peerPublicKey: this._peerPublicKey
-    })
+      this._peerAccountName = 'client'
+      this._accountName = 'server'
+      this._authToken = opts.token
+      this._prefix = opts.prefix
+    } else {
+      // deprecate this version?
+      assertOptionType(opts, 'secret', 'string')
+      assertOptionType(opts, 'peerPublicKey', 'string')
+
+      this._secret = opts.secret
+      this._peerAccountName = opts.peerPublicKey
+      this._accountName = Token.publicKey(this._secret)
+      this._prefix = Token.prefix({
+        secretKey: this._secret,
+        peerPublicKey: this._peerAccountName,
+        currencyCode: this._currencyCode,
+        currencyScale: this._currencyScale
+      })
+
+      // Token uses ECDH to get the ledger prefix from secret and public key
+      this._authToken = Token.authToken({
+        secretKey: this._secret,
+        peerPublicKey: this._peerAccountName
+      })
+
+      if (opts.prefix && opts.prefix !== this._prefix) {
+        throw new InvalidFieldsError('invalid prefix. got "' +
+          opts.prefix + '", expected "' + this._prefix + '"')
+      }
+    }
 
-    this._prefix = Token.prefix({
-      secretKey: this._secret,
-      peerPublicKey: this._peerPublicKey,
-      currencyCode: this._currencyCode,
-      currencyScale: this._currencyScale
-    })
+    this._stateful = !!opts._store
```

Comment:
I think there should be an explicit option to use the plugin in asymmetric (client) mode so that if your code has a bug where it's not passing in the store you won't accidentally run it in an in-memory-only mode.